main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintSolver.scala

```diff
@@ -272,6 +272,10 @@ object ConstraintSolver {
       resolveEquality(t1, t2, prov, renv, constr0.loc).map {
         case ResolutionResult(subst, constrs, p) => ResolutionResult(subst @@ subst0, constrs, progress = p)
       }
+    case TypeConstraint.EqJvmMethod(mvar, tpe, method, tpes, prov) =>
+      resolveEquality(mvar, tpe, subst0(prov), renv, constr0.loc)
```

Comment:
Here we should be able to call your resolve method more or less directly.

We will just need to fiddle a bit with the return type.

Reply:
Indeed. Now it becomes simple and nice.